src/watchmaker/workers/salt.py

```diff
@@ -661,6 +667,49 @@ def _set_grain(self, grain, value):
         self.log.info('Setting grain `%s` ...', grain)
         super(SaltLinux, self)._set_grain(grain, value)
 
+    def _install_pip_modules(self):
+        if self.pip_install:
+            python = self.run_salt(['grains.get', 'pythonexecutable'])
+            py_exec = python['stdout'].decode().split('\n')[1].strip()
+            self.log.info('Salt Python interpreter: `%s`', py_exec)
+
+            try:
+                ensure = self.call_process([py_exec, '-m', 'ensurepip'])
+                self.log.info('Ensure pip: %s', ensure['stdout'])
+                ver = self.call_process([py_exec, '-m', 'pip', '--version'])
+                self.log.info('Pip version: %s', ver['stdout'])
+            except WatchmakerException:
+                self.log.info('Pip not installed for Salt interpreter!')
+                self.log.info('Attempting pip install...')
+                get_pip = [
+                    'curl',
+                    'https://bootstrap.pypa.io/get-pip.py',
+                    '-o',
+                    'get-pip.py'
```

Comment:
where does this file end up? we should make sure we clean up.

Reply:
I'm not sure `self.call_process()` can handle it, but it may be cleaner to pipe curl to python... i.e. `curl https://<get-pip.py> | <py_exec> -`

Reply:
Hmm, this might still get caught up in environment issues. See how we were setting the index url for the get-pip.py execution previously (via userdata)... https://github.com/plus3it/watchmaker/pull/843/files